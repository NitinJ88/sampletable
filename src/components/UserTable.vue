<template>
    <b-table
      hoverable
      :sticky-header="true"
      :data="data"  
      :columns="columns"        
    >     
        <b-table-column
          centered                       
          v-slot="props"          
        >        
          <span >{{ props.row.id}}</span>
        </b-table-column>
        <b-table-column          
          v-slot="props"
          class="users-table--b-table--email"               
        >                                 
          <span>{{ props.row.email }}</span>            
        </b-table-column>  
         <b-table-column          
          v-slot="props"               
        >     
         <div
            v-if="IsAdminUser"
            class="users-table--b-table--remove-user"            
            @click="removeUser(props.row)"
          >                            
          <remove-user />        
         </div>   
        </b-table-column>                            
    </b-table>
</template>

<script>
import removeUser from "../assets/remove_user.svg";
export default {
    name: 'UserTable',
     components: { removeUser },
     props:{
        
        isAdminUser: {
        type: Boolean,
        default: false
        }
     },
    data() {
        return {
           data : [
    {
        id: 1,
        email: "abc@gmail.com"
    },
     {
        id: 2,
        email: "xyz@gmail.com"
    },
    {
        id: 3,
        email: "pqr@yahoo.com"
    },               
],             
            columns: [
                    {
                        field: 'id',
                        label: 'ID',                        
                        numeric: true
                    },
                    {
                        field: 'email',
                        label: 'email',
                    },
                    {
                        field: 'manage',
                        label: 'Manage',
                    }    
            ]   
        }
    },
    computed: {
    IsAdminUser() {     
      return this.isAdminUser;
    },
    methods: {
    removeUser(member) {     
      this.$emit("remove-user", member);
    }
    }
  }    
};
</script>
<style lang="scss" scoped>
.users-table--b-table--remove-user {
  vertical-align: middle;
}
.users-table--b-table--email {
  display: inline-block;
  padding-left: 0.5em;
  text-overflow: ellipsis;
  max-width: 73vw;
  overflow: hidden;
  vertical-align: middle;
  padding-bottom: 0.2em;
}

</style>