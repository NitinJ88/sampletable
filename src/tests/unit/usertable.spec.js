import { createLocalVue, mount } from "@vue/test-utils";
import UserTable from "@/components/UserTable.vue";
import Buefy from "buefy";

function getWrapper() {
    return mount (UserTable, {
      localVue,
      propsData: {
        isAdminUser: true
      },
      mocks: {
       $i18n: { locale: "test" }        
      }      
    });    
}

const localVue = createLocalVue();
localVue.use(Buefy);
describe("UserTable.vue", () => {
test("Test UserTable", async () => {
    const wrapper = getWrapper();    
    await wrapper.vm.$nextTick();
    expect(wrapper.find("users-table--b-table--email").exists()).toBe(
      true
    );
    expect(wrapper.element).toMatchSnapshot(); 
});

});