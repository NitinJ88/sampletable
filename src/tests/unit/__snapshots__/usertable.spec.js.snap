// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTable.vue Test UserTable 1`] = `
<div
  class="b-table"
>
  <!---->
   
  <!---->
   
  <!---->
  <!---->
  <!---->
  <div
    class="table-wrapper has-mobile-cards has-sticky-header"
  >
    <table
      class="table is-hoverable"
    >
      <thead>
        <tr>
          <!---->
          <!---->
          <th
            class=""
          >
            <div
              class="th-wrap is-numeric"
            >
              <span
                class="is-relative"
              >
                 ID 
                <span
                  class="icon sort-icon is-small is-invisible"
                >
                  <i
                    class="mdi mdi-arrow-up"
                  />
                </span>
              </span>
            </div>
          </th>
          <th
            class=""
          >
            <div
              class="th-wrap"
            >
              <span
                class="is-relative"
              >
                 email 
                <span
                  class="icon sort-icon is-small is-invisible"
                >
                  <i
                    class="mdi mdi-arrow-up"
                  />
                </span>
              </span>
            </div>
          </th>
          <th
            class=""
          >
            <div
              class="th-wrap"
            >
              <span
                class="is-relative"
              >
                 Manage 
                <span
                  class="icon sort-icon is-small is-invisible"
                >
                  <i
                    class="mdi mdi-arrow-up"
                  />
                </span>
              </span>
            </div>
          </th>
          <!---->
        </tr>
        <!---->
        <!---->
      </thead>
      <tbody>
        <tr
          class=""
          draggable="false"
        >
          <!---->
          <!---->
          <td
            class="has-text-right"
            data-label="ID"
          >
            <span>
              1
            </span>
          </td>
          <td
            class=""
            data-label="email"
          >
            <span>
              abc@gmail.com
            </span>
          </td>
          <td
            class=""
            data-label="Manage"
          >
            <span />
          </td>
          <!---->
        </tr>
        <!---->
        <!---->
        <tr
          class=""
          draggable="false"
        >
          <!---->
          <!---->
          <td
            class="has-text-right"
            data-label="ID"
          >
            <span>
              2
            </span>
          </td>
          <td
            class=""
            data-label="email"
          >
            <span>
              xyz@gmail.com
            </span>
          </td>
          <td
            class=""
            data-label="Manage"
          >
            <span />
          </td>
          <!---->
        </tr>
        <!---->
        <!---->
        <tr
          class=""
          draggable="false"
        >
          <!---->
          <!---->
          <td
            class="has-text-right"
            data-label="ID"
          >
            <span>
              3
            </span>
          </td>
          <td
            class=""
            data-label="email"
          >
            <span>
              pqr@yahoo.com
            </span>
          </td>
          <td
            class=""
            data-label="Manage"
          >
            <span />
          </td>
          <!---->
        </tr>
        <!---->
        <!---->
        <!---->
      </tbody>
      <!---->
    </table>
    <!---->
  </div>
  <!---->
</div>
`;
